pkg/promlib/models/scope_test.go

```diff
@@ -92,7 +92,105 @@ func TestApplyQueryFilters(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			expr, err := ApplyQueryFilters(tt.query, tt.scopeFilters, tt.adhocFilters)
+			expr, err := ApplyFiltersAndGroupBy(tt.query, tt.scopeFilters, tt.adhocFilters, nil)
+
+			if tt.expectErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, expr, tt.expected)
+			}
+		})
+	}
+}
+
+func TestApplyQueryFiltersAndGroupBy_GroupBy(t *testing.T) {
+	tests := []struct {
+		name      string
+		query     string
+		groupBy   []string
+		expected  string
+		expectErr bool
+	}{
+		{
+			name:      "GroupBy with no aggregate expression",
+			groupBy:   []string{"job"},
+			query:     `http_requests_total`,
+			expected:  `http_requests_total`,
```

Comment:
I think the expected result here would be `sum by (job) (http_requests_total)`. But up for discussion and we can also consider queries without sum functions a next step.